owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/files/DownloadRemoteFileOperation.kt

```diff
@@ -0,0 +1,182 @@
+            result = RemoteOperationResult(e)
+            Timber.e(e, "Download of $remotePath to $tmpPath: ${result.logMessage}")
+        }
+        return result
+    }
+
+    @Throws(Exception::class)
+    private fun downloadFile(client: OwnCloudClient, targetFile: File): RemoteOperationResult<Unit> {
+        val result: RemoteOperationResult<Unit>
+        var it: Iterator<OnDatatransferProgressListener>
+        var fos: FileOutputStream? = null
+        var bis: BufferedInputStream? = null
+        var savedFile = false
+
+        val getMethod = GetMethod(URL(client.userFilesWebDavUri.toString() + WebdavUtils.encodePath(remotePath)))
+
+        try {
+            val status = client.executeHttpMethod(getMethod)
+
+            if (isSuccess(status)) {
+                targetFile.createNewFile()
+                bis = BufferedInputStream(getMethod.getResponseBodyAsStream())
+                fos = FileOutputStream(targetFile)
+                var transferred: Long = 0
+                val contentLength = getMethod.getResponseHeader(HttpConstants.CONTENT_LENGTH_HEADER)
+                val totalToTransfer =
+                    if (contentLength != null && contentLength.isNotEmpty()) contentLength.toLong() else 0
+                val bytes = ByteArray(4096)
+                var readResult: Int
+                while (bis.read(bytes).also { readResult = it } != -1) {
+                    synchronized(mCancellationRequested) {
+                        if (mCancellationRequested.get()) {
+                            getMethod.abort()
+                            throw OperationCancelledException()
+                        }
+                    }
+                    fos.write(bytes, 0, readResult)
+                    transferred += readResult.toLong()
+                    synchronized(mDataTransferListeners) {
+                        it = mDataTransferListeners.iterator()
+                        while (it.hasNext()) {
+                            it.next().onTransferProgress(
+                                readResult.toLong(), transferred, totalToTransfer,
+                                targetFile.name
+                            )
+                        }
+                    }
+                }
+
+                if (transferred == totalToTransfer) {  // Check if the file is completed
+                    savedFile = true
+                    val modificationTime =
+                        getMethod.getResponseHeaders()?.get("Last-Modified")
+                            ?: getMethod.getResponseHeader("last-modified")
+
+                    if (modificationTime != null) {
+                        val modificationDate = WebdavUtils.parseResponseDate(modificationTime)
+                        modificationTimestamp = modificationDate?.time ?: 0
+                    } else {
+                        Timber.e("Could not read modification time from response downloading %s", remotePath)
+                    }
```

Comment:
```suggestion
                    modificationTime?.let {
                        val modificationDate = WebdavUtils.parseResponseDate(modificationTime)
                        modificationTimestamp = modificationDate?.time ?: 0
                    } ?: Timber.e("Could not read modification time from response downloading %s", remotePath)
```

Reply:
Lastly, I prefer the `if else` clause instead of the `let ?:` one. It is easier to follow from my point of view.